src/window/context_settings.rs

```diff
@@ -42,6 +41,21 @@ pub const CONTEXT_DEBUG: u32 = 1 << 2;
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Copy)]
 pub struct ContextSettings(pub ffi::sfContextSettings);
 
+impl ContextSettings {
+    /// Creates a new ContextSettings with all parameters 
+    /// default, except `antialiasing_level`
+    pub fn new_with_antialiasing_level(aa_level: u32) -> ContextSettings {
```

Comment:
I added a TODO saying exactly that in my local copy of the repo after
sending the PR. Why don't you let me do it?

I'll submit a new, updated PR today then.

Thanks for the feedback.

On Tue, Mar 29, 2016, 1:29 PM Mika Attila <notifications@github.com> wrote:

> In src/window/context_settings.rs
> <https://github.com/jeremyletang/rust-sfml/pull/116#discussion_r57684585>:
>
> > @@ -42,6 +41,21 @@ pub const CONTEXT_DEBUG: u32 = 1 << 2;
> >  #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Copy)]
> >  pub struct ContextSettings(pub ffi::sfContextSettings);
> >
> > +impl ContextSettings {
> > +    /// Creates a new ContextSettings with all parameters
> > +    /// default, except `antialiasing_level`
> > +    pub fn new_with_antialiasing_level(aa_level: u32) -> ContextSettings {
>
> I don't think adding a new_with_foo function for every property we want
> to set is a composable solution.
>
> We could instead support the builder pattern
> <https://doc.rust-lang.org/book/method-syntax.html#builder-pattern>, so
> the user could do e.g. let ctx =
> ContextSettings::default().antialiasing(8);.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/jeremyletang/rust-sfml/pull/116/files/cdcd37534e9b7160995a9dbe9287d728249dba8c#r57684585>
>